### PR TITLE
Call validation.json endpoint as an API

### LINKED ISSUE #1627: Fetch validation results as an API call
When https://github.com/mozilla/addons-server/pull/14717 lands, which should fix https://github.com/mozilla/addons-server/issues/14655, which currently breaks code manager, we will need to change the logic we use to fetch validation results. 

Currently we simply fetch a URL, bit now we need to make an API call, which will require some logic changes in code-manager code.

### PR DESCRIPTION
**Pull request by bobsilverberg**
Fixes #1627

### CODE REVIEW

**Comment by bobsilverberg** [comment]
@willdurand This relates to a P1, so needs to be looked at ASAP. Sorry for this. I'm going to land the related change on server soon, so you should be able to test this against the dev server once that lands.

File: src/api/index.tsx [hunk 1]
```diff
@@ -542,3 +543,15 @@ export const deleteComment = async ({
...
+
+type GetValidationParams = {
+  apiState: ApiState;
+  url: string;
+};
+
+export const getValidation = async ({ apiState, url }: GetValidationParams) => {
+  return callApi<ResponseOnly<ExternalLinterResult>>({
+    apiState,
+    endpointUrl: url,
```
Comment (willdurand) [inline src/api/index.tsx:555]: Can we construct the endpoint here instead? We don't need to pass the `url` anymore I think. We can probably pass the addon ID.

**Comment by willdurand** [comment]
I haven't tried the patch but it looks good :) If you land the api patch, I can give it a try

File: src/api/index.tsx (hunk 1 above)
Comment (bobsilverberg) [reply src/api/index.tsx:555]: We need the file ID as well, but yes, we should be able to do this. I'll file a follow-up.

**Comment by bobsilverberg** [comment]
> I haven't tried the patch but it looks good :) If you land the api patch, I can give it a try

I've landed the API patch, but I'm not sure if it's been deployed yet.

**Pull request merged**